core/primitives/src/reed_solomon.rs

```diff
@@ -12,24 +13,32 @@ const RS_TTL: u64 = 2 * 1024;
 /// reed solomon instead to work around the memory leak in reed solomon
 /// implementation <https://github.com/darrenldl/reed-solomon-erasure/issues/74>
 pub struct ReedSolomonWrapper {
-    rs: ReedSolomon,
-    ttl: u64,
+    rs: RefCell<ReedSolomon>,
+    ttl: RefCell<u64>,
```

Comment:
I think it might be a good idea to have a single block of RefCell or whatever is needed based on the zulip discussion instead of two different blocks.  To ensure that you do not have to take two different "locks" potentially seeing corrupted data in between.

Reply:
Not required any more as we just got rid of the whole wrapper :)